packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -162,6 +160,14 @@ export class TokenDetectionController extends StaticIntervalPollingController<
 
   #networkClientId: NetworkClientId;
 
+  #chainIdAgainstWhichToDetect: Hex;
+
+  #addressAgainstWhichToDetect: string;
+
+  #networkClientIdAgainstWhichToDetect: NetworkClientId;
```

Comment:
Should these be maintained separately from `#chainId`, `#selectedAddress`, `#networkClientId`?

Reply:
You're right that consumers should be able to call `detectTokens` independently, and it should just work. 

It makes sense that we would cache the currently selected network as well as the currently selected account. I don't think we need to cache the chain ID, because we can always ask the network client.

As for these three variables, though, it seems to me that we've created them so that we can break up `detectTokens` into pieces. But, why not pass along these values, then? It seems that this would make it easier for someone to read the code because there would be fewer things for them to track.

Reply:
Agree that a pure style that avoids using class fields might be more readable here.

Reply:
Removed three unnecessary class fields here: [`ab0e3eb` (#3938)](https://github.com/MetaMask/core/pull/3938/commits/ab0e3eb1baee597aa371b1e71b101d9a3a08cdd0)

Reply:
Also removed `#chainId`: [`12ba664` (#3938)](https://github.com/MetaMask/core/pull/3938/commits/12ba6647b4655b950bd05854fd97c992a835cbe7)